backend/src/services/auto_trader.py

```
@@ -120,151 +109,169 @@ async def _scan_and_buy(self):
                     if sell_result.get("success"):
-                        logger.info(f"AutoTrader: Successfully sold {details['symbol']}. Transaction: {sell_result.get('transaction_id')}")
                         tokens_to_remove.append(token_address)
                         if self.socketio:
-                            self.socketio.emit('auto_trade_event', {'type': 'sell', 'token': details['symbol'], 'reason': 'stop_loss', 'status': 'success'})
-                    else:
-                        logger.error(f"AutoTrader: Failed to sell {details['symbol']}: {sell_result.get('message')}")
-                        if self.socketio:
-                            self.socketio.emit('auto_trade_event', {'type': 'sell', 'token': details['symbol'], 'reason': 'stop_loss', 'status': 'failed', 'error': sell_result.get('message')})
+                            self.socketio.emit('auto_trade_event', {'type': 'sell', 'token': details['token_symbol'], 'reason': reason, 'status': 'success'})
 
             except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)